monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,389 @@
+    """
+    This class generates a set of bundles according to the cross-validation folds, each of them can run independently.
+
+    Args:
+        algo: a dictionary that outlines the algorithm to use.
+        algo_path: the directory path to save the algorithm templates. Default is the current working dir.
+        data_stats_filename: the path to the data stats file (generated by DataAnalyzer)
+        data_src_cfg_name: the path to the data source file, which contains the data list
+
+    .. code-block:: bash
+
+        python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
+    """
+
+    def __init__(self, algo_path: str = ".", algos=None, data_stats_filename=None, data_src_cfg_name=None):
+        self.algos: Any = []
+
+        if algos is None:
+            # trigger the download process
+            zip_download_dir = TemporaryDirectory()
+            algo_compressed_file = os.path.join(zip_download_dir.name, "algo_templates.tar.gz")
+            download_and_extract(algo_zip, algo_compressed_file, algo_path, md5)
+            zip_download_dir.cleanup()
+            sys.path.insert(0, os.path.join(algo_path, "algo_templates"))
+            algos = copy(default_algos)
+            for name in algos:
+                algos[name]["template_configs"] = os.path.join(algo_path, default_algos[name]["template_configs"])
+                algos[name]["scripts_path"] = os.path.join(algo_path, default_algos[name]["scripts_path"])
+
+        if isinstance(algos, dict):
+            for algo_name, algo_params in algos.items():
+                self.algos.append(ConfigParser(algo_params).get_parsed_content())
+                self.algos[-1].name = algo_name
+        else:
+            self.algos = ensure_tuple(algos)
+
+        self.data_stats_filename = data_stats_filename
+        self.data_src_cfg_filename = data_src_cfg_name
+        self.history: List[Dict] = []
+
+    def set_data_stats(self, data_stats_filename: str):  # type: ignore
+        """
+        Set the data stats filename
+
+        Args:
+            data_stats_filename: filename of datastats
+        """
+        self.data_stats_filename = data_stats_filename
+
+    def get_data_stats(self):
+        """Get the filename of the data stats"""
+        return self.data_stats_filename
+
+    def set_data_src(self, data_src_cfg_filename):
+        """
+        Set the data source filename
+
+        Args:
+            data_src_cfg_filename: filename of data_source file
+        """
+        self.data_src_cfg_filename = data_src_cfg_filename
+
+    def get_data_src(self, fold_idx=0):
+        """Get the data source filename"""
+        return self.data_src_cfg_filename
+
+    def get_history(self, *args, **kwargs) -> List:
+        """get the history of the bundleAlgo object with their names/identifiers"""
+        return self.history
+
+    def generate(self, output_folder=".", fold_idx: Union[int, Sequence[int]] = 0):
```

Comment:
it is better to have number of folds as input here. for now, we are experimenting with 5-fold. users may use it for different number of folds.

Reply:
Fixed. Thanks!